### PR TITLE
configure -tls-server-name when global.cloud.enabled=true so that it matches the server certificate minted from HCP

### PR DESCRIPTION
**Pull request by jmurret**
Changes proposed in this PR:
- configure -tls-server-name for consul-k8s components that use connection manager when global.cloud.enabled = true so that it matches the server certificate minted from HCP 
-

How I've tested this PR:
- bats tests
- manual integration testing with HCP

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by ndhanushkodi — APPROVED** [review]
Maybe at some point we could standardize the flag name to either -tls-server-name or -consul-tls-server-name across all the commands but that's definitely not blocking for this fix!

**Review by curtbushko — APPROVED** [review]

File: charts/consul/templates/mesh-gateway-deployment.yaml [hunk 1]
```diff
@@ -213,6 +213,8 @@ spec:
             {{- end }}
             {{- if and .Values.externalServers.enabled .Values.externalServers.tlsServerName }}
             -tls-server-name={{.Values.externalServers.tlsServerName }} \
```
Comment (thisisnotashwin) [inline charts/consul/templates/mesh-gateway-deployment.yaml:215]: Should this be `consul-tls-server-name` as well?

**Review by thisisnotashwin — APPROVED** [review]
As Nitya suggested, it is a little unclear where we use `consul-tls-server-name` and `tls-server-name` and we should put something in the backlog to clear that up.

File: charts/consul/templates/mesh-gateway-deployment.yaml (hunk 1 above)
Comment (kschoche) [reply charts/consul/templates/mesh-gateway-deployment.yaml:215]: I think this should be `-tls-server-name`  for `consul-dataplane` and `-consul-tls-server-name` for other commands.
We should probably clear this up though.

File: charts/consul/templates/mesh-gateway-deployment.yaml [hunk 2]
```diff
@@ -213,6 +213,8 @@ spec:
...
+            {{- else if .Values.global.cloud.enabled }}
+            -consul-tls-server-name=server.{{ .Values.global.datacenter}}.{{ .Values.global.domain}} \
```
Comment (kschoche) [inline charts/consul/templates/mesh-gateway-deployment.yaml:217]:
```suggestion
            -tls-server-name=server.{{ .Values.global.datacenter}}.{{ .Values.global.domain}} \
```
I believe this is `-tls-server-name` ?

**Review by kschoche — CHANGES_REQUESTED** [review]
I'm not sure the flags are all being passed correctly, happy to pair and take a look with you, but feel free to ignore if I'm actually wrong :)

One other thing, non-blocking, since this is repeated verbatim in so many places, should we move it into a template in `_helpers.tpl`?

```
                {{- if and $root.Values.externalServers.enabled $root.Values.externalServers.tlsServerName }}
                -tls-server-name={{$root.Values.externalServers.tlsServerName }} \
                {{- else if $root.Values.global.cloud.enabled }}
                -consul-tls-server-name=server.{{ $root.Values.global.datacenter}}.{{ $root.Values.global.domain}} \
```

File: charts/consul/templates/mesh-gateway-deployment.yaml (hunk 1 above)
Comment (jmurret) [reply charts/consul/templates/mesh-gateway-deployment.yaml:215]: I just updated this to use tls-server-name everywhere.  I think differentiating for dataplane vs other components is more of an implementation detail, where this setting in every case is meant to set the TLSConfig.

File: charts/consul/templates/mesh-gateway-deployment.yaml (hunk 2 above)
Comment (jmurret) [reply charts/consul/templates/mesh-gateway-deployment.yaml]: this has been fixed.

File: control-plane/version/version.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ var (
 	// A pre-release marker for the version. If this is "" (empty string)
 	// then it means that it is a final release. Otherwise, this is a pre-release
 	// such as "dev" (in development), "beta", "rc1", etc.
-	VersionPrerelease = "dev"
+	VersionPrerelease = "beta1"
```
Comment (thisisnotashwin) [inline control-plane/version/version.go:22]: Is this correct? I think this should still be `dev`?

File: cli/version/version.go [hunk 4]
```diff
@@ -19,7 +19,7 @@ var (
 	// A pre-release marker for the version. If this is "" (empty string)
 	// then it means that it is a final release. Otherwise, this is a pre-release
 	// such as "dev" (in development), "beta", "rc1", etc.
-	VersionPrerelease = "dev"
+	VersionPrerelease = "beta1"
```
Comment (thisisnotashwin) [inline cli/version/version.go:22]: this should be `dev` too

**Pull request merged**